src/twisted/internet/iocpreactor/abstract.py

```diff
@@ -188,48 +188,49 @@ def _resumeWriting(self):
 
 
     def _cbWrite(self, rc, numBytesWritten, evt):
-        if self._handleWrite(rc, numBytesWritten, evt):
+        if self._handleWrite(rc, numBytesWritten, evt) and \
+        self._writeScheduled is None:
             self.doWrite()
 
 
-    def _handleWrite(self, rc, numBytesWritten, evt):
+    def _handleWrite(self, rc, numBytesWritten, evt, updateOffset=False):
```

Comment:
Please add docstring for the new updateOffset argument, even if this is a private method.